request/executor.go

```diff
@@ -6,45 +6,90 @@ import (
 	"net/http"
 )
 
+const maxRedirect = 10
+
 // ExecuteRequest executes an HTTP request based on the specified options.
-func ExecuteRequest(request Request, profileNames []string, variables map[string]string) (*ExecutedRequestResponse, error) {
-	httpRequest, configuredRequest, httpRequestErr := BuildRequest(request, profileNames, variables)
-	if httpRequestErr != nil {
-		return nil, httpRequestErr
-	}
+func ExecuteRequest(request Request, profileNames []string, variables map[string]string) ([]ExecutedRequestResponse, error) {
+	requestsToExecute := make([]Request, 1)
+	requestsToExecute[0] = request
+	redirectCount := 0
 
-	client := &http.Client{}
-	httpResponse, httpResponseErr := client.Do(httpRequest)
-	if httpResponseErr != nil {
-		return nil, httpResponseErr
+	client := &http.Client{
+		// Do not auto-follow redirects
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
 	}
 
-	bodyBytes, readErr := ioutil.ReadAll(httpResponse.Body)
+	result := make([]ExecutedRequestResponse, 0)
 
-	if readErr != nil {
-		return nil, readErr
-	}
+	for {
+		currentRequest := requestsToExecute[0]
+		requestsToExecute = requestsToExecute[1:]
 
-	body := ""
-	if len(bodyBytes) != 0 {
-		body = string(bodyBytes)
-	}
+		httpRequest, currentConfiguredRequest, httpRequestErr := BuildRequest(currentRequest, profileNames, variables)
+		if httpRequestErr != nil {
+			return nil, httpRequestErr
+		}
 
-	headers := make(map[string][]string)
-	for k, vs := range httpResponse.Header {
-		headers[k] = append(headers[k], vs...)
-	}
+		httpResponse, httpResponseErr := client.Do(httpRequest)
+		if httpResponseErr != nil {
+			return nil, httpResponseErr
+		}
+
+		bodyBytes, readErr := ioutil.ReadAll(httpResponse.Body)
+
+		if readErr != nil {
+			return nil, readErr
+		}
+
+		headers := make(map[string][]string)
+		for k, vs := range httpResponse.Header {
+			headers[k] = append(headers[k], vs...)
+		}
 
-	response := Response{
-		StatusCode: httpResponse.StatusCode,
-		Status:     httpResponse.Status,
-		Headers:    headers,
-		Body:       body,
-		Protocol:   fmt.Sprintf("%d.%d", httpResponse.ProtoMajor, httpResponse.ProtoMinor),
+		response := Response{
+			StatusCode: httpResponse.StatusCode,
+			Status:     httpResponse.Status,
+			Headers:    headers,
+			Body:       string(bodyBytes),
+			Protocol:   fmt.Sprintf("%d.%d", httpResponse.ProtoMajor, httpResponse.ProtoMinor),
+		}
+
+		result = append(result, ExecutedRequestResponse{
+			Request:  *currentConfiguredRequest,
+			Response: response,
+		})
+
+		if response.StatusCode == http.StatusMovedPermanently || response.StatusCode == http.StatusFound || response.StatusCode == http.StatusSeeOther {
```

Comment:
Nit: you could consider moving this logic to it's own boolean helper (like `shouldRedirect`).